TelePharo-Playground.package/TlpRemoteDoItReceiver.class/instance/compiler.st

```diff
@@ -0,0 +1,4 @@
+accessing
+compiler
+	"Shameless hack to get at the CompilerContext of the remote system for help with remote CompletionContext"
+	^ self remotePharo evaluate: [ nil compiler ]
```

Comment:
Could you explain why we need a compiler from the remote side? 
Can we use "self class compiler" instead?

One of ideas was to be able to work with very tiny images which does not have even a compiler.
But I appreciate that in current state that could be a simplest fix for you problem

Reply:
What is important is to get a compiler for the class that you look at. Else it can not highlight instance variables or class variables.

Reply:
Thank you both for the feedback. Initially, this method wasn't present and was the source of the stack trace in Pharo 8 and I did stumble across `self class compilerClass` while trying to solve the more advanced issue of the auto-complete context. If you use a local compiler on the client side, it doesn't get the `environment` (a Dictionary of all the classes and methods) from the remote side.

My test for this was to create a class only on the remote side and try to get it to show up during auto-complete.

In my exploration of standard, non-TelePharo behavior of auto-complete in the Playground, it wasn't actually binding to a specific class compiler, but using the compiler of the nil class, so that's why I thought it was probably safe to do so.  I'm thinking that perhaps if you're working in the context of an editor _IN_ the browser, it may possibly bind to the class you're actually working on, but that wasn't one of the things I was testing.

Reply:
> If you use a local compiler on the client side, it doesn't get the `environment` (a Dictionary of all the classes and methods) from the remote side.

Yes. The trick to set the remote environment (available in #remotePharo instance) into the local compiler. It is how the remote inspector implements remote doIts.
We can merge your version and experiment later but maybe you can try it at this step